sdk/pinocchio/src/sysvars/mod.rs

```diff
@@ -76,7 +92,10 @@ pub unsafe fn get_sysvar_unchecked(
 
         match result {
             crate::SUCCESS => Ok(()),
-            e => Err(e.into()),
+            OFFSET_LENGTH_EXCEEDS_SYSVAR => Err(ProgramError::InvalidArgument),
+            SYSVAR_NOT_FOUND => Err(ProgramError::UnsupportedSysvar),
```

Comment:
@ilya-bobyr Simplified the match arm since clippy complained when I made a similar change on the SDK – likely we don't have all warnings enabled in this repo, but since we should fix that it is better to update the code.